app/containers/HomePage/index.js

```diff
@@ -130,4 +131,4 @@ export default compose(
   withReducer,
   withSaga,
   withConnect,
-)(HomePage);
+)(toJS(HomePage));
```

Comment:
According to that article, selectors should always return Immutable objects. The article suggests wrapping "dumb" components in the `toJS` HOC so that they are more pure, easier to test, etc. 

It goes on to say that if you are having performance issues, this may not be the best solution, so perhaps we can solve this with a comment? Something like...

*If your component has many rerenders that cannot be avoided, you might improve performance by removing the `toJS` HOC and using Immutable.JS in your dumb components. This does, however, have the drawback of making your dumb components less portable and less testable.*

Reply:
cool, I will take a look on document

Reply:
@et let's use our existing composition chains to wrap components, eg...

```js
// containers/HomePage.js
import withImmutable from 'utils/toJS';
// ...
export default compose(
  withReducer,
  withSaga,
  withConnect,
  withImmutable,
)(HomePage);
```

this is the same as `toJS(HomePage)`—functional associativity FTW

Reply:
@justingreenberg that's good idea and I like the `compose` in place rather than just exporting left to right.